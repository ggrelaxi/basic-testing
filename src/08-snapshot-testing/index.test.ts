// Uncomment the code below and write your tests
import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  // Check match by expect(...).toStrictEqual(...)
  test('should generate linked list from values 1', () => {
    // Write your test here
    expect(generateLinkedList([])).toStrictEqual({ value: null, next: null });
    expect(generateLinkedList([1])).toStrictEqual({
      value: 1,
      next: { value: null, next: null },
    });
    expect(generateLinkedList([1, 2])).toStrictEqual({
      value: 1,
      next: { value: 2, next: { value: null, next: null } },
    });
  });

  // Check match by comparison with snapshot
  test('should generate linked list from values 2', () => {
    // Write your test here
    expect(generateLinkedList([])).toMatchSnapshot();
    expect(generateLinkedList([1])).toMatchSnapshot();
    expect(generateLinkedList([1, 2])).toMatchSnapshot();
  });
});
