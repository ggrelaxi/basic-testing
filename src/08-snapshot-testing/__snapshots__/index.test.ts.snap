// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": null,
  "value": null,
}
`;

exports[`generateLinkedList should generate linked list from values 2 2`] = `
{
  "next": {
    "next": null,
    "value": null,
  },
  "value": 1,
}
`;

exports[`generateLinkedList should generate linked list from values 2 3`] = `
{
  "next": {
    "next": {
      "next": null,
      "value": null,
    },
    "value": 2,
  },
  "value": 1,
}
`;
